### PR TITLE
[keepalived-vip]: Add support for proxy protocol

### PR DESCRIPTION
**Pull request by aledbf**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/2041)

### CODE REVIEW

**Comment by saracen** [comment]
@aledbf Using the `aledbf/kube-keepalived-vip:0.12` image I get the following error as soon as the container starts when I enable proxy mode.

```
I1210 18:48:09.746535       1 keepalived.go:183] cleaning ipvs configuration
F1210 18:48:09.749330       1 utils.go:430] unexpected error writing haproxy.cfg: open /etc/haproxy/haproxy.cfg: no such file or directory

```

**Comment by sandys** [comment]
hi guys - we are looking for this support because we want to pass through SSL as well as preserve source-ip to the pods. Can I request someone to review and move this ahead please ?
@bprashanth @aledbf

**Comment by aledbf** [comment]
@sandys while this is merged you can use `aledbf/kube-keepalived-vip:0.13` (contains the changes in this PR)

**Comment by sandys** [comment]
@aledbf - thanks a lot ! Do you have a sample config to do ssl passthrough
and source IP preservation ?

Would be extremely helpful!

On Dec 24, 2016 19:01, "Manuel Alejandro de Brito Fontes" <
notifications@github.com> wrote:

@sandys <https://github.com/sandys> while this is merged you can use
aledbf/kube-keepalived-vip:0.13 (contains the changes in this PR)

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/contrib/pull/2041#issuecomment-269084540>,
or mute the thread

**Comment by aledbf** [comment]
> Do you have a sample config to do ssl passthrough and source IP preservation ?

yes in the readme of this PR :)

https://github.com/kubernetes/contrib/blob/8950925d6cf043fb4ef34913fda3b14f45aa8795/keepalived-vip/README.md#proxy-protocol-mode

File: keepalived-vip/keepalived.tmpl [hunk 1]
```diff
@@ -27,8 +32,17 @@ vrrp_instance vips {
...
+{{ if .proxyMode}}
+  # In proxy mode there is no need to create virtual servers
+  track_script {
+    chk_haproxy weight 1
+  }
+{{ end }}
+
 }
 
+{{ if not .proxyMode }}
```
Comment (saracen) [inline keepalived-vip/keepalived.tmpl]: This means that **every** service listed in the configmap will use a forwarding type of PROXY, even if "DR" is explicitly stated on any other service. Is this the intended behaviour?

**Pull request closed without merging**